controllers/etcd_controller.go

```diff
@@ -852,8 +865,11 @@ func (r *EtcdReconciler) getMapFromEtcd(etcd *druidv1alpha1.Etcd) (map[string]in
 	}
 
 	var statefulsetReplicas int
-	if etcd.Spec.Replicas != 0 {
-		statefulsetReplicas = 1
+	// Check if Spec.Replicas is odd or even
+	if etcd.Spec.Replicas&1 == 1 {
+		statefulsetReplicas = etcd.Spec.Replicas
+	} else {
+		return map[string]interface{}{}, fmt.Errorf("Spec.Replicas should not be even number: %d", etcd.Spec.Replicas)
```

Comment:
Should we set this error on the `Etcd` resource status somewhere? Otherwise, this message will be lost in the logs.

Reply:
The error returned this way is already set as error status [https://github.com/gardener/etcd-druid/blob/048277792bde4401801303aa62d177d10fcaa243/controllers/etcd_controller.go#L257](url)